### PR TITLE
To macro for rows

### PR DESCRIPTION
**Pull request by b3liever**
This adds a de-serialization macro for seq[string] useful with db_*.nim row iterator and parsecsv

### CODE REVIEW

**Comment by Araq** [comment]
Don't include the 'zip macro' commit in this PR please. This needs to have longer name than ``to`` IMO and it should be in its own module.

In the implementation use the newer ``move`` for strings.

**Comment by b3liever** [comment]
> This needs to have longer name than ``to`` IMO and it should be in its own module.

So as I understand you, you are not really fond of another ``to`` macro in the stdlib and its fine by me.
Since today I discovered custom annotations, I plan to develop this further. So a nimble package would be better. Btw the ``json.to`` macro needs some love and if only there was a bounty posted I could rewrite and fix it...

**Comment by Araq** [comment]
> Btw the json.to macro needs some love and if only there was a bounty posted I could rewrite and fix it...

Well create an issue and then we add a bounty to it.

**Comment by Araq** [comment]
Can this be moved to ``db_common`` or maybe a new module, ``db_rows``?

File: lib/pure/sugar.nim [hunk 1]
```diff
@@ -238,3 +238,91 @@ macro distinctBase*(T: typedesc): untyped =
...
+    type
+      Vector2D = tuple[x, y: int]
+      Date = distinct string
+      Entity = object
+        x, y: float
+        z: Vector2D
+        w: Date
+    proc `==`(a, b: Date): bool = string(a) == string(b)
+    var row = @["1.0", "2.0", "3", "4", "2018-01-10"]
+    assert row.to(Entity) == Entity(x: 1.0, y: 2.0, z: (3, 4),
```
Comment (krux02) [inline lib/pure/sugar.nim]: This unrolling of the tuple is really questionable to me. Is there a reason why this is done?

**Comment by krux02** [comment]
Is this still active?

**Comment by krux02** [comment]
This behavior is highly questionable. It is inactive and I don't want it. Can this be closed?

**Comment by b3liever** [comment]
> This behavior is highly questionable. It is inactive and I don't want it. Can this be closed?

If you refer to my behavior (just say it, we are all friends) there's nothing questionable, I am sick and afk since last Tuesday. Plus I mentioned that ``I plan to develop this further`` You can close this issue, many thanks!

File: lib/pure/sugar.nim (hunk 1 above)
Comment (b3liever) [reply lib/pure/sugar.nim]: bc of the recursion

**Comment by krux02** [comment]
@b3liever no it is not about you. Sorry if you felt that way. I am referring to the implicit unrolling of tuple element. I really don't like that part.

**Pull request closed without merging**